adsplanetnamepipe/collect.py

```diff
@@ -0,0 +1,186 @@
+from typing import List, Tuple
+
+from adsplanetnamepipe.models import KnowledgeBase, KnowledgeBaseHistory
+from adsplanetnamepipe.utils.common import EntityArgs, Synonyms
+from adsplanetnamepipe.utils.search_retrieval import SearchRetrieval
+from adsplanetnamepipe.utils.match_excerpt import MatchExcerpt
+from adsplanetnamepipe.utils.extract_keywords import ExtractKeywords
+from adsplanetnamepipe.utils.astrobert_ner import AstroBERTNER
+from adsplanetnamepipe.utils.local_llm import LocalLLM
+from adsplanetnamepipe.utils.paper_relevance import PaperRelevance
+
+
+class CollectKnowldegeBase():
+
+    search_retrieval = None
+
+    def __init__(self, args: EntityArgs):
+        """
+
+        :param args:
+        """
+        self.args = args
+        # step 1 of the pipeline
+        self.search_retrieval = SearchRetrieval(args)
+        # step 2 of the pipeline
+        self.match_excerpt = MatchExcerpt(args)
+        # step 3 of the pipeline
+        self.astrobert_ner = AstroBERTNER(args)
+        # step 4 of the pipeline
+        self.extract_keywords = ExtractKeywords(args)
+        # step 5b of the pipeline, only scoring the positive records
+        self.paper_relevance = PaperRelevance(args)
+        # step 5c of the pipeline, only scoring the positive records
+        self.local_llm = LocalLLM(args)
+        self.vocabulary = Synonyms().add_synonyms([args.target,
+                                                   args.feature_type, args.feature_type_plural,
+                                                   args.feature_name])
+
+    def get_paper_relevance_score(self, doc: dict) -> float:
+        """
+
+        :param doc:
+        :return:
+        """
+        text = ' '.join(doc.get('title', '')) + ' ' + doc.get('abstract', '') + ' ' + doc.get('body', '')
+        return self.paper_relevance.forward(
+            text,
+            bibstem=doc.get('bibcode', '')[4:9].strip('.'),
```

Comment:
Yes, not sure how do we want to proceed here. Need to chat about this further.

Reply:
I'd leave it as-is for now, since we won't be moving away from bibcodes for a bit longer yet, but maybe make an issue on the repo to handle it later.

Reply:
I also brought this up and i was told I should not be concerned, since probably this issue will not be solved anytime soon, and I probably retire before needed to be dealing with this, lol.

Reply:
I also brought this up and was told that I shouldn’t be concerned, as this issue likely won’t be resolved anytime soon, and I’ll probably retire before needing to deal with it, lol.